### DISCUSSION TITLE
How to use the VS Designer? Always getting errors

### DISCUSSION THREAD

**Discussion by davidgiga1993** [gitextensions/gitextensions#9376] [Q&A] [answered]
Hi,
I'm trying to edit some GUI items but no matter which control I open in the editor I get the following exception:
```
at GitUI.ThreadHelper.ThrowIfNotOnUIThread(String callerMemberName) in D:\Development\Projects\gitextensions\GitExtUtils\GitUI\ThreadHelper.cs:line 59
at ResourceManager.GitExtensionsControlInitialiser..ctor(GitExtensionsControl control) in D:\Development\Projects\gitextensions\ResourceManager\GitExtensionsControlInitialiser.cs:line 25
at ResourceManager.GitExtensionsControl..ctor() in D:\Development\Projects\gitextensions\ResourceManager\GitExtensionsControl.cs:line 20
at GitUI.GitModuleControl..ctor() in D:\Development\Projects\gitextensions\GitUI\GitModuleControl.cs:line 108  
```
Looks like the condition to check if the code is executed by the designer doesn't work as expected?
Or am I missing something here?

I'm using a fresh VS2019 install with the .net desktop development feature installed

**Comment by mstv** [comment, marked as answer]
The .NET 5.0 support has been promised for VS2019 17.
For the time being, you could design (and work) on the branch `release/3.5` and optionally cherry-pick it to `master`.

**Reply by RussKie** [reply]
FWIW it is working, https://github.com/gitextensions/gitextensions/pull/9372 is updated in the designer.
However you do need to comment out `ThreadHelper.ThrowIfNotOnUIThread` call in GitExtensionsControlInitialiser for that. I haven't had a chance to establish whether it is a regression in the new designer or not.

**Reply by mstv** [reply]
Indeed, `ThreadHelper.ThrowIfNotOnUIThread` is causing this since the migration to .NET 5.0.

The check for the designer seems to not work anymore:
```
        public static void ThrowIfNotOnUIThread([CallerMemberName] string callerMemberName = "")
        {
            if (LicenseManager.UsageMode == LicenseUsageMode.Designtime)
            {
                return;
            }

            if (!JoinableTaskContext.IsOnMainThread)
```

**Reply by RussKie** [reply]
Yes, the designer is run in a separate process, so this check now fails. There's a suggestion on a possible workaround in https://github.com/dotnet/winforms/issues/4146#issuecomment-716239006, but I don't have cycles to test it.

**Reply by RussKie** [reply]
@davidgiga1993 I opened https://github.com/gitextensions/gitextensions/pull/9387, it should make things much-much easier.
